ProChess/Shared/Rules/InCheckPossibility.cs

```diff
@@ -0,0 +1,380 @@
+﻿using Microsoft.VisualBasic;
+using Shared.Data;
+using Shared.Helpers.Extensions;
+using System.Collections.Generic;
+using System.Data.Common;
+
+namespace Shared.Rules
+{
+    public class InCheckPossibility
+    {
+        private readonly int[] _directionOffsets = { 1, -1, 2, -2 };
+        private readonly int[] _edgeBoard = { 0, 7 };
+
+        public List<Cell> CouldPossibleCellsCreateCheck(Piece? activePiece, List<Cell> cellsPossible, List<Piece> whitePieces, List<Piece> blackPieces)
```

Comment:
I saw that the first parameter is not nullable. Why "activePiece" is it declared nullable?